mlflow/models/evaluation/default_evaluator.py

```diff
@@ -459,12 +459,17 @@ def _extract_output_and_other_columns(model_predictions, output_column_name):
             other_output_columns = pd.DataFrame(
                 [{k: v for k, v in p.items() if k != output_column_name} for p in model_predictions]
             )
-        elif len(model_predictions) > 1:
-            if output_column_name is None:
-                raise MlflowException(
-                    ERROR_MISSING_OUTPUT_COLUMN_NAME,
-                    error_code=INVALID_PARAMETER_VALUE,
-                )
+        elif len(model_predictions[0]) == 1:
+            # Set the only key as self.predictions and its value as self.y_pred
+            key, value = list(model_predictions[0].items())[0]
+            y_pred = pd.Series(value, name=key)
+            output_column_name = key
+        elif output_column_name is None:
+            raise MlflowException(
+                ERROR_MISSING_OUTPUT_COLUMN_NAME,
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+        else:
```

Comment:
Can we test the following cases: (some might be covered already, but just listing here to double check)
1. Model output has 1 column "col1", predictions=None: logged table shows "col1"
2. Model output has 1 column "col1", predictions="col1": logged table shows "col1"
3. Model output has 1 column "col1", predictions="col2" # Should throw error
4. Model output has 2 columns "col1", "outputs", predictions=None # Should throw error, not using default "outputs"
5. Model output has 2 columns "col1", "outputs", predictions="col2" # Should throw error
6. Model output has 2 columns "col1", "outputs", predictions="col1": logged table shows "col1"

Reply:
Yes we have test all the above usecase.

Except 6.
Model output has 2 columns "col1", "outputs", predictions="col1": logged table shows "col1" and "outputs" since outputs could be additional context or something that we should log as well. But yes all the above cases are covered :) 
Thanks for listing them out